src/lib/storage/free_disk_spaces_list.rb

```diff
@@ -20,54 +20,36 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
-require "storage"
+require "storage/free_disk_space"
 require "storage/disk_size"
+require "storage/devices_list"
 require "storage/refinements/disk"
 
 module Yast
   module Storage
-    # Prototype of a class to allow querying a devigraph for its elements
-    #
-    # The class is now used in the RSpec tests and in the proposal code but the
-    # API will likely change a lot depending on the discussion on
-    # https://lists.opensuse.org/yast-devel/2016-03/msg00053.html
-    # Thus the lack of documentation for the concrete methods
-    class DevicegraphQuery
+    class FreeDiskSpacesList < DevicesList
+      list_of FreeDiskSpace
+
       using Refinements::Disk
 
       # Free disk space below this size will be disregarded
       TINY_FREE_CHUNK = DiskSize.MiB(30)
 
-      attr_reader :devicegraph
-
-      def initialize(devicegraph, disk_names: nil)
-        @devicegraph = devicegraph
-        @disk_names = disk_names
-      end
-
-      def disks
-        if @disk_names
-          @disk_names.map { |disk_name| ::Storage::Disk.find(@devicegraph, disk_name) }
-        else
-          devicegraph.all_disks.to_a
-        end
+      def useful
```

Comment:
at least below when it is used for me reasonable_big look there better, but I welcome any better name.